genesyscloud/group/data_source_genesyscloud_group.go

```diff
@@ -11,47 +11,26 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/mypurecloud/platform-client-sdk-go/v125/platformclientv2"
 )
 
-func DataSourceGroup() *schema.Resource {
-	return &schema.Resource{
-		Description: "Data source for Genesys Cloud Groups. Select a group by name.",
-		ReadContext: provider.ReadWithPooledClient(dataSourceGroupRead),
-		Schema: map[string]*schema.Schema{
-			"name": {
-				Description: "Group name.",
-				Type:        schema.TypeString,
-				Required:    true,
-			},
-		},
-	}
-}
-
 func dataSourceGroupRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
-	groupsAPI := platformclientv2.NewGroupsApiWithConfig(sdkConfig)
+	gp := getGroupProxy(sdkConfig)
 
-	exactSearchType := "EXACT"
-	nameField := "name"
 	nameStr := d.Get("name").(string)
 
-	searchCriteria := platformclientv2.Groupsearchcriteria{
-		VarType: &exactSearchType,
-		Value:   &nameStr,
-		Fields:  &[]string{nameField},
-	}
-
 	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
-		groups, _, getErr := groupsAPI.PostGroupsSearch(platformclientv2.Groupsearchrequest{
-			Query: &[]platformclientv2.Groupsearchcriteria{searchCriteria},
-		})
+		groups, _, getErr := gp.getGroupsByName(ctx, nameStr)
 		if getErr != nil {
 			return retry.NonRetryableError(fmt.Errorf("Error requesting group %s: %s", nameStr, getErr))
 		}
 
+		if *groups.Total > 1 {
```

Comment:
I added this.  This was a potential bug because if we ever got back more then one group we would always grab the first one and this could be wrong